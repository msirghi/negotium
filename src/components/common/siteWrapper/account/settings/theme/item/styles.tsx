import { makeStyles } from '@mui/styles';
import colors from '../../../../../../../common/styles/colors';

export const useThemeItemStyles = makeStyles({
  container: {
    width: 185,
    border: `1px solid ${colors.greys['300']}`,
    borderRadius: 6,
    cursor: 'pointer',
    transition: 'transform 250ms',
    '&:hover': {
      transform: 'translateY(-2px)',
    },
  },
  title: {
    borderTopLeftRadius: 5,
    borderTopRightRadius: 5,
    backgroundColor: (props: { color?: string, textColor?: string }) => props.color,
    color: (props: { textColor?: string, color?: string }) => props.textColor || colors.white,
    height: 35,
    padding: 6,
    fontSize: 14,
  },
  skeleton: {
    height: 75,
    padding: 5,
  },
  textSkeleton: {
    borderRadius: 1,
    marginLeft: 5,
    marginBottom: 1,
  },
  checkIcon: {
    marginLeft: 'auto',
    marginRight: 5,
  },
});
