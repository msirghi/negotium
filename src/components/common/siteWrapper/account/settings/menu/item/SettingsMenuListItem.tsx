import { OverridableComponent } from '@mui/types';
import { SvgIconTypeMap } from '@mui/material';
import { FC } from 'react';
import { makeStyles } from '@mui/styles';
import { Row } from '../../../../../utilities/row/Row';
import colors from '../../../../../../../common/styles/colors';
import { useIsMobile } from '../../../../../../../common/hooks/common/useIsMobile';
import { If } from '../../../../../utilities/if/If';

type Props = {
  Icon: OverridableComponent<SvgIconTypeMap> & { muiName: string };
  title: string;
  isActive: boolean;
  onClick: () => void;
};

const useStyles = makeStyles({
  container: (props: { isMobile: boolean; isActive: boolean }) => ({
    marginTop: 10,
    transition: '.3s background-color ease-in-out',
    padding: '10px 15px',
    borderRadius: 5,
    width: props.isMobile ? 'initial' : '220px !important',
    '&:hover': {
      backgroundColor: colors.greys['200'],
      cursor: 'pointer',
    },
    backgroundColor: props.isActive ? colors.greys['300'] : '',
  }),
  title: {
    marginLeft: 5,
  },
});

export const SettingsMenuListItem: FC<Props> = ({
  Icon,
  onClick,
  isActive,
  title,
}) => {
  const isMobile = useIsMobile();
  const classes = useStyles({ isActive, isMobile });

  return (
    <Row alignVerticalCenter className={classes.container} onClick={onClick}>
      <Icon fontSize={'small'} />
      <If condition={!isMobile}>
        <div id={`${title}-item`} className={classes.title}>{title}</div>
      </If>
    </Row>
  );
};
