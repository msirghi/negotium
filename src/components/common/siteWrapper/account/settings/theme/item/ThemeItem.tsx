import { FC } from 'react';
import { useThemeItemStyles } from './styles';
import { ThemeBodySkeleton } from '../skeleton/ThemeBodySkeleton';
import CheckIcon from '@mui/icons-material/Check';
import { Row } from '../../../../../utilities/row/Row';
import { If } from '../../../../../utilities/if/If';
import { SiteTheme } from '../../../../../../../common/constants/types';

type Props = {
  theme: SiteTheme;
  selected?: boolean;
  onClick: (theme: string) => void;
};

export const ThemeItem: FC<Props> = ({ theme, selected, onClick }) => {
  const { color, label, textColor } = theme;
  const classes = useThemeItemStyles({ color, textColor });

  const handleClick = () => {
    onClick(theme.internalKey);
  };

  return (
    <div
      className={classes.container}
      id={`theme-${label}`}
      onClick={handleClick}
    >
      <div>
        <div className={classes.title}>{label}</div>
        <div className={classes.skeleton}>
          <ThemeBodySkeleton />
          <ThemeBodySkeleton />
          <ThemeBodySkeleton />
          <If condition={selected}>
            <Row fullWidth>
              <CheckIcon fontSize={'small'} className={classes.checkIcon} />
            </Row>
          </If>
        </div>
      </div>
    </div>
  );
};
