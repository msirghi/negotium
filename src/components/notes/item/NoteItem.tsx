import { Note } from '../../../common/types/notes.types';
import { FC, useState } from 'react';
import {
  Card,
  CardActions,
  CardContent,
  IconButton,
  Tooltip,
  Typography,
} from '@mui/material';
import { useNoteItemStyles } from './styles';
import { NoteDialog } from './dialog/NoteDialog';
import SlateUtils from '../../../common/utils/slateUtils';
import DeleteOutlineIcon from '@mui/icons-material/DeleteOutline';
import DateUtils from '../../../common/utils/dateUtils';
import { NOTE_DATE_FORMAT } from '../../../common/constants/constants';
import useTranslation from 'next-translate/useTranslation';

type Props = {
  note: Note;
  onNoteRemove: (id: Note['id']) => void;
  onNoteUpdate: (note: Note) => void;
};

export const NoteItem: FC<Props> = ({ note, onNoteRemove, onNoteUpdate }) => {
  const { title, description, id, createdDate } = note;
  const classes = useNoteItemStyles();
  const { t } = useTranslation('notes');

  const [dialogOpen, setDialogOpen] = useState(false);

  const openDialog = () => {
    setDialogOpen(true);
  };

  const closeDialog = () => {
    setDialogOpen(false);
  };

  const onDeleteClick = (e: MouseEvent) => {
    e.stopPropagation();
    onNoteRemove(id);
  };

  return (
    <>
      <NoteDialog
        open={dialogOpen}
        note={note}
        closeDialog={closeDialog}
        onNoteUpdate={onNoteUpdate}
      />

      <Card className={classes.container} onClick={openDialog}>
        <CardContent>
          <Typography variant="h5" component="div" className={classes.title}>
            {title}
          </Typography>

          <Typography
            className={classes.description}
            color="text.secondary"
            gutterBottom
          >
            {SlateUtils.serialize(
              SlateUtils.getInitialValueForSlate(description)
            )}
          </Typography>
        </CardContent>
        <CardActions>
          <div className={classes.date}>
            {DateUtils.formatDate(createdDate, NOTE_DATE_FORMAT)}
          </div>
          {/*// @ts-ignore*!/*/}
          <IconButton className={classes.deleteButton} onClick={onDeleteClick} data-testid={'delete-note-btn'}>
            <Tooltip title={t('deleteNote')!}>
              <DeleteOutlineIcon />
            </Tooltip>
          </IconButton>
        </CardActions>
      </Card>
    </>
  );
};
