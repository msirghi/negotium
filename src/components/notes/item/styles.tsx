import { makeStyles } from '@mui/styles';

export const useNoteItemStyles = makeStyles({
  container: {
    cursor: 'pointer',
    height: '100%',
    boxShadow: '1px 3px 5px 1px rgba(34, 60, 80, 0.13);',
  },
  description: {
    fontSize: 14
  }
});
