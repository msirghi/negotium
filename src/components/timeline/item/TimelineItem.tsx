import { Task } from '../../../common/types/tasks.types';
import { FC } from 'react';
import { useTimelineItemStyles } from './styles';
import DateUtils from '../../../common/utils/dateUtils';
import { Box, useTheme } from '@mui/system';
import { Checkbox } from '@mui/material';
import { Row } from '../../common/utilities/row/Row';

type Props = {
  task: Task;
  onClick: (task: Task) => void;
  active: boolean;
  markAsDone: (id: Task['id']) => void;
};

export const TimelineItem: FC<Props> = ({ task, onClick, active, markAsDone }) => {
  const theme = useTheme();
  const date = DateUtils.formatDateForTask(task.dueDate);
  const defaultColor = theme.palette.custom.mainBackgroundColor;
  const activeColor = theme.palette.custom.activeMainBackgroundColor;
  const classes = useTimelineItemStyles({ active, activeColor, defaultColor });

  const handleClick = () => onClick(task);

  const handleMarkAsDone = () => markAsDone(task.id);

  return (
    <div className={classes.container} onClick={handleClick}>
      <Row alignVerticalCenter>
        <Checkbox checked={task.completed} color={'warning'} size={'small'} className={classes.checkbox} onChange={handleMarkAsDone} />
        <Box>
          <p className={classes.title}>{task.title}</p>
          {date.value && <p className={classes.date}>{date.value}</p>}
        </Box>
      </Row>
    </div>
  );
};
