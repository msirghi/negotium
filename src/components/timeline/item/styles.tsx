import { makeStyles } from '@mui/styles';
import colors from '../../../common/styles/colors';

export const useTimelineItemStyles = makeStyles({
  container: (props: { active: boolean; defaultColor: string; activeColor: string }) => {
    return {
      backgroundColor: props.active ? props.activeColor : props.defaultColor,
      transition: 'all .3s',
      fontSize: 14,
      borderRadius: 10,
      color: colors.white,
      padding: '0.1px 10px',
      marginTop: 5,
      cursor: 'pointer',
    };
  },
  title: {},
  date: {
    marginTop: -5,
    fontSize: 12,
  },
});
