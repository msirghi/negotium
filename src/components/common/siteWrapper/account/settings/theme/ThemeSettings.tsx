import { ThemeItem } from './item/ThemeItem';
import { siteThemes } from '../../../../../../common/constants/constants';
import { Grid } from '@mui/material';
import { useThemeSettingsStyles } from './styles';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from '../../../../../../redux/store';
import { setSiteTheme } from '../../../../../../redux/account/accountSlice';
import AccountService from '../../../../../../services/AccountService';

export const ThemeSettings = () => {
  const classes = useThemeSettingsStyles();
  const siteTheme = useSelector(
    (state: RootState) => state.account.metadata.theme
  );
  const dispatch = useDispatch();

  const onThemeChange = async (theme: string) => {
    if (siteTheme !== theme) {
      dispatch(setSiteTheme(theme));
      await AccountService.updateUserTheme(theme);
    }
  };

  return (
    <div>
      <div className={classes.title}>
        Choose a theme that matches your preference.
      </div>
      <Grid container spacing={3}>
        {siteThemes.map((theme) => (
          <Grid key={theme.color} item>
            <ThemeItem
              theme={theme}
              selected={siteTheme === theme.internalKey}
              onClick={onThemeChange}
            />
          </Grid>
        ))}
      </Grid>
    </div>
  );
};
