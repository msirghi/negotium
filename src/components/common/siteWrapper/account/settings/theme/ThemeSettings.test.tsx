import { MockReduxProvider } from '../../../../../../common/tests/TestUtils';
import { ThemeSettings } from './ThemeSettings';
import renderer, { act } from 'react-test-renderer';
import { reduxStoreMock } from '../../../../../../common/tests/mockData/redux-store-mock';
import AccountService from '../../../../../../services/AccountService';
import { mount } from 'enzyme';
import { ThemeItem } from './item/ThemeItem';

describe('ThemeSettings', () => {
  const renderComponent = () => {
    return (
      <MockReduxProvider reduxStore={reduxStoreMock}>
        <ThemeSettings />
      </MockReduxProvider>
    );
  };

  beforeEach(() => {
    AccountService.updateUserTheme = jest.fn();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const tree = renderer.create(renderComponent());
    expect(tree).toMatchSnapshot();
  });

  it('should not make api call to update theme if same theme is selected', () => {
    const wrapper = mount(renderComponent());
    const themeItem = wrapper.find(ThemeItem).at(0);

    act(() => {
      themeItem.props().onClick(reduxStoreMock.account.metadata.theme);
    });
    expect(AccountService.updateUserTheme).not.toBeCalled();
  });

  it('should make an api call to update theme if another theme is selected', () => {
    const wrapper = mount(renderComponent());
    const themeItem = wrapper.find(ThemeItem).at(0);

    act(() => {
      themeItem.props().onClick('red');
    });
    expect(AccountService.updateUserTheme).toBeCalled();
  });
});
