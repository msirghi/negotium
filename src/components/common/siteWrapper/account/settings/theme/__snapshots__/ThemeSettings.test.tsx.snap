// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSettings should match the snapshot 1`] = `
<div>
  <div
    className="makeStyles-title-1"
  >
    Choose a theme that matches your preference.
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        className="makeStyles-container-2"
        id="theme-Default"
        onClick={[Function]}
      >
        <div>
          <div
            className="makeStyles-title-3 makeStyles-title-7"
          >
            Default
          </div>
          <div
            className="makeStyles-skeleton-4"
          >
            <div
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "width": "initial",
                }
              }
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular css-14wtl70-MuiSkeleton-root"
                style={
                  Object {
                    "height": 9,
                    "width": 10,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text makeStyles-textSkeleton-5 css-1l6af8c-MuiSkeleton-root"
                style={
                  Object {
                    "height": 15,
                    "width": "100%",
                  }
                }
              />
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "width": "initial",
                }
              }
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular css-14wtl70-MuiSkeleton-root"
                style={
                  Object {
                    "height": 9,
                    "width": 10,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text makeStyles-textSkeleton-5 css-1l6af8c-MuiSkeleton-root"
                style={
                  Object {
                    "height": 15,
                    "width": "100%",
                  }
                }
              />
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "width": "initial",
                }
              }
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular css-14wtl70-MuiSkeleton-root"
                style={
                  Object {
                    "height": 9,
                    "width": 10,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text makeStyles-textSkeleton-5 css-1l6af8c-MuiSkeleton-root"
                style={
                  Object {
                    "height": 15,
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        className="makeStyles-container-2"
        id="theme-Noir"
        onClick={[Function]}
      >
        <div>
          <div
            className="makeStyles-title-3 makeStyles-title-11"
          >
            Noir
          </div>
          <div
            className="makeStyles-skeleton-4"
          >
            <div
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "width": "initial",
                }
              }
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular css-14wtl70-MuiSkeleton-root"
                style={
                  Object {
                    "height": 9,
                    "width": 10,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text makeStyles-textSkeleton-5 css-1l6af8c-MuiSkeleton-root"
                style={
                  Object {
                    "height": 15,
                    "width": "100%",
                  }
                }
              />
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "width": "initial",
                }
              }
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular css-14wtl70-MuiSkeleton-root"
                style={
                  Object {
                    "height": 9,
                    "width": 10,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text makeStyles-textSkeleton-5 css-1l6af8c-MuiSkeleton-root"
                style={
                  Object {
                    "height": 15,
                    "width": "100%",
                  }
                }
              />
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "width": "initial",
                }
              }
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular css-14wtl70-MuiSkeleton-root"
                style={
                  Object {
                    "height": 9,
                    "width": 10,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text makeStyles-textSkeleton-5 css-1l6af8c-MuiSkeleton-root"
                style={
                  Object {
                    "height": 15,
                    "width": "100%",
                  }
                }
              />
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "flex-start",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "width": "100%",
                }
              }
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall makeStyles-checkIcon-6 css-ptiqhd-MuiSvgIcon-root"
                data-testid="CheckIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        className="makeStyles-container-2"
        id="theme-Light Grey"
        onClick={[Function]}
      >
        <div>
          <div
            className="makeStyles-title-3 makeStyles-title-15"
          >
            Light Grey
          </div>
          <div
            className="makeStyles-skeleton-4"
          >
            <div
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "width": "initial",
                }
              }
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular css-14wtl70-MuiSkeleton-root"
                style={
                  Object {
                    "height": 9,
                    "width": 10,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text makeStyles-textSkeleton-5 css-1l6af8c-MuiSkeleton-root"
                style={
                  Object {
                    "height": 15,
                    "width": "100%",
                  }
                }
              />
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "width": "initial",
                }
              }
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular css-14wtl70-MuiSkeleton-root"
                style={
                  Object {
                    "height": 9,
                    "width": 10,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text makeStyles-textSkeleton-5 css-1l6af8c-MuiSkeleton-root"
                style={
                  Object {
                    "height": 15,
                    "width": "100%",
                  }
                }
              />
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "width": "initial",
                }
              }
            >
              <span
                className="MuiSkeleton-root MuiSkeleton-rectangular css-14wtl70-MuiSkeleton-root"
                style={
                  Object {
                    "height": 9,
                    "width": 10,
                  }
                }
              />
              <span
                className="MuiSkeleton-root MuiSkeleton-text makeStyles-textSkeleton-5 css-1l6af8c-MuiSkeleton-root"
                style={
                  Object {
                    "height": 15,
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
